lib/engine/game/base.rb

```diff
@@ -907,6 +907,7 @@ def end_game!
         @round_counter += 1
         scores = result.map { |name, value| "#{name} (#{format_currency(value)})" }
         @log << "-- Game over: #{scores.join(', ')} --"
+        @round.steps.find { |s| s.is_a?(Engine::Step::Message) }.unpass!
```

Comment:
can you just make message step unpassable???

Reply:
how can someone even pass message?

Reply:
On this branch, it happens in game 16904. Actually it happens that way on master as well every time the active corporation closes, but usually the next one becomes active right away so it doesn't matter (on this branch and master, `force_next_entity!`, called [here](https://github.com/tobymao/18xx/blob/c069a914d0e9bedf222a0139e7868467a700f068/lib/engine/game/base.rb#L1308) passes all the steps [here](https://github.com/tobymao/18xx/blob/c069a914d0e9bedf222a0139e7868467a700f068/lib/engine/round/operating.rb#L49))

* the bankruptcy leaves one active player, so it ends the game
* the active corporation also closes during this bankruptcy--at the point where the president must declare bankruptcy, the active corp's price is $10; as part of going bankrupt, the president sells shares they're normally not allowed to sell, thus dropping the price to $0 and closing it
* `force_next_entity!` passes the step
* the game is over, so there's no new entity/round/step to be active

I'll change it to make message unpassable, that sounds better, and goes along with how Message is also unskippable

Reply:
Fixed in 16904